dbms/tests/integration/test_storage_s3/test.py

```diff
@@ -1,159 +1,168 @@
+                                        "Action": "s3:GetObject",
+                                        "Resource": "arn:aws:s3:::root/*"
+                                    },
+                                    {
+                                        "Sid": "",
+                                        "Effect": "Allow",
+                                        "Principal": {"AWS": "*"},
+                                        "Action": "s3:PutObject",
+                                        "Resource": "arn:aws:s3:::root/*"
+                                    }
+                                ]}
+
+    minio_client.set_bucket_policy(cluster.minio_bucket, json.dumps(bucket_read_write_policy))
+
+
+# Returns content of given S3 file as string.
+def get_s3_file_content(cluster, filename):
+    # type: (ClickHouseCluster, str) -> str
+
+    data = cluster.minio_client.get_object(cluster.minio_bucket, filename)
+    data_str = ""
+    for chunk in data.stream():
+        data_str += chunk
+    return data_str
 
 
 @pytest.fixture(scope="module")
-def started_cluster():
+def cluster():
     try:
         cluster = ClickHouseCluster(__file__)
-        instance = cluster.add_instance("dummy")
+        cluster.add_instance("dummy", with_minio=True)
+        logging.info("Starting cluster...")
         cluster.start()
+        logging.info("Cluster started")
 
-        cluster.communication_port = 10000
-        instance.copy_file_to_container(os.path.join(os.path.dirname(__file__), "server.py"), "test_server.py")
-        cluster.bucket = "abc"
-        instance.exec_in_container(["python", "test_server.py", str(cluster.communication_port), cluster.bucket], detach=True)
-        cluster.mock_host = instance.ip_address
-
-        for i in range(10):
-            try:
-                data = get_communication_data(cluster)
-                cluster.redirecting_to_http_port = data["redirecting_to_http_port"]
-                cluster.preserving_data_port = data["preserving_data_port"]
-                cluster.multipart_preserving_data_port = data["multipart_preserving_data_port"]
-                cluster.redirecting_preserving_data_port = data["redirecting_preserving_data_port"]
-            except:
-                logging.error(traceback.format_exc())
-                time.sleep(0.5)
-            else:
-                break
-        else:
-            assert False, "Could not initialize mock server"
+        prepare_s3_bucket(cluster)
+        logging.info("S3 bucket created")
 
         yield cluster
-
     finally:
         cluster.shutdown()
 
 
 def run_query(instance, query, stdin=None, settings=None):
+    # type: (ClickHouseInstance, str, object, dict) -> str
+
     logging.info("Running query '{}'...".format(query))
     result = instance.query(query, stdin=stdin, settings=settings)
     logging.info("Query finished")
-    return result
 
+    return result
 
-def test_get_with_redirect(started_cluster):
```

Comment:
Can you check that GET with redirect works?

Reply:
This is tested at `test_put_get_with_redirect`